edit/PersistentVolumeClaim.vue

```diff
@@ -0,0 +1,104 @@
+<script>
+import LabeledSelect from '@/components/form/LabeledSelect';
+import UnitInput from '@/components/form/UnitInput';
+import RadioGroup from '@/components/form/RadioGroup';
+import Checkbox from '@/components/form/Checkbox';
+import LabeledInput from '@/components/form/LabeledInput';
+import { mapGetters } from 'vuex';
+import { removeObject } from '@/utils/array';
+import { STORAGE_CLASS, PV } from '@/config/types';
+import { allHash } from '@/utils/promise';
+export default {
+
+  components: {
+    LabeledSelect, UnitInput, RadioGroup, Checkbox, LabeledInput
+  },
+  props:      {
+    mode: {
+      type:    String,
+      default: 'create'
+    },
+    value: {
+      type:    Object,
+      default: () => {
+        return {};
+      }
+    },
+    registerBeforeHook: {
+      type:    Function,
+      default: null,
+    },
+  },
+  async fetch() {
+    const hash = await allHash({
+      storageClasses:    this.$store.dispatch('cluster/findAll', { type: STORAGE_CLASS }),
+      persistentVolumes: this.$store.dispatch('cluster/findAll', { type: PV }),
+    });
+
+    this.storageClasses = hash.storageClasses.map(sc => sc.metadata.name);
+    this.persistentVolumes = hash.persistentVolumes.map(pv => pv.metadata.name);
+  },
```

Comment:
Save the result of findAll to data and make the mapping a computed property, so that they update when resources come and go with change events.

```
async fetch() {
  this.allStorageClasses = await ...findAll...
  this.allPersistentVolumes = await ...
},

computed: {
  storageClassNames() {
    return this.allStorageClasses.map(x => x.name);
  }
}
```